sklearn/semi_supervised/tests/test_label_propagation.py

```diff
@@ -59,3 +62,15 @@ def test_predict_proba():
         clf = estimator(**parameters).fit(samples, labels)
         assert_array_almost_equal(clf.predict_proba([[1., 1.]]),
                                   np.array([[0.5, 0.5]]))
+
+
+def test_alpha_deprecation():
+    X, y = make_classification(n_samples=100)
+    y[::3] = -1
+    lp_default = label_propagation.LabelPropagation()
+    lp_default_y = assert_no_warnings(lp_default.fit, X, y).transduction_
```

Comment:
This is getting a warning (although not the one we're concerned about) and the assertion fails:
```

======================================================================
FAIL: sklearn.semi_supervised.tests.test_label_propagation.test_alpha_deprecation
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/scikit-learn/scikit-learn/sklearn/semi_supervised/tests/test_label_propagation.py", line 71, in test_alpha_deprecation
    lp_default_y = assert_no_warnings(lp_default.fit, X, y).transduction_
  File "/home/travis/build/scikit-learn/scikit-learn/sklearn/utils/testing.py", line 230, in assert_no_warnings
    ', '.join(str(warning) for warning in w)))
AssertionError: Got warnings when calling fit: [{message : RuntimeWarning('underflow encountered in exp',), category : 'RuntimeWarning', filename : '/home/travis/build/scikit-learn/scikit-learn/sklearn/metrics/pairwise.py', lineno : 837, line : None}]
```

I've realised that I didn't fix the `random_state` in `make_classification`. We should do that. But I also haven't investigated why we should be getting exp underflow. Do you know?